CalPatRec/src/HelixFinder_module.cc

```diff
@@ -0,0 +1,2559 @@
+//-----------------------------------------------------------------------------
+const XYZVectorF* HelixFinder::getPos(size_t& tcHitsIndex) {
+
+  int hitIndice = _tcHits[tcHitsIndex].hitIndice;
+
+  if (hitIndice >= 0) {
+    return &_chColl->at(hitIndice).pos();
+  }
+  if (hitIndice == -2) {
+    return &_stopTargPos;
+  }
+  if (hitIndice == -1) {
+    return &_caloPos;
+  }
+
+  else {
+    return nullptr;
+  }
+}
+
+//-----------------------------------------------------------------------------
+// compute distance between two points in XY plane
+//-----------------------------------------------------------------------------
+float HelixFinder::computeXYplaneDist2(const XYZVectorF& p1, const XYZVectorF& p2) {
+
+  return (p1.x() - p2.x()) * (p1.x() - p2.x()) + (p1.y() - p2.y()) * (p1.y() - p2.y());
+}
+
+//-----------------------------------------------------------------------------
+// updates circleError of hit given some circle parameters
+//-----------------------------------------------------------------------------
+void HelixFinder::computeCircleError2(size_t& tcHitsIndex, float& xC, float& yC) {
+
+  int hitIndice = _tcHits[tcHitsIndex].hitIndice;
+
+  if (hitIndice == -1) {
+    _tcHits[tcHitsIndex].circleError2 = _caloClusterSigma * _caloClusterSigma;
+  } else {
+    float transVar = _chColl->at(hitIndice).transVar();
+    float x = getPos(tcHitsIndex)->x();
+    float y = getPos(tcHitsIndex)->y();
+    float dx = x - xC;
+    float dy = y - yC;
+    float dxn = dx * _chColl->at(hitIndice).vDir().x() + dy * _chColl->at(hitIndice).vDir().y();
+    float costh2 = dxn * dxn / (dx * dx + dy * dy);
+    float sinth2 = 1 - costh2;
+    _tcHits[tcHitsIndex].circleError2 =
+        _chColl->at(hitIndice).wireVar() * sinth2 + transVar * costh2;
+    // float xP = getPos(tcHitsIndex)->x();
+    // float yP = getPos(tcHitsIndex)->y();
+    // float radVecX = (xP-xC)/std::sqrt((xP-xC)*(xP-xC)+(yP-yC)*(yP-yC));
+    // float radVecY = (yP-yC)/std::sqrt((xP-xC)*(xP-xC)+(yP-yC)*(yP-yC));
+    // float wireErr = _chColl->at(hitIndice).wireRes();
+    // float wireVecX = _chColl->at(hitIndice).uDir().x();
+    // float wireVecY = _chColl->at(hitIndice).uDir().y();
+    // float projWireErr = wireErr*(wireVecX*radVecX + wireVecY*radVecY);
+    // float transErr = _chColl->at(hitIndice).transRes();
+    // float transVecX = _chColl->at(hitIndice).uDir().y();
+    // float transVecY = -_chColl->at(hitIndice).uDir().x();
+    // float projTransErr = transErr*(transVecX*radVecX + transVecY*radVecY);
+    // _tcHits[tcHitsIndex].circleError2 = projTransErr*projTransErr+projWireErr*projWireErr;
+  }
+}
+
+//-----------------------------------------------------------------------------
+// compute residual between point an circle
+//-----------------------------------------------------------------------------
+float HelixFinder::computeCircleResidual2(size_t& tcHitsIndex, float& xC, float& yC, float& rC) {
+
```

Comment:
Please consider using XYVectorF in this and similar interfaces, and in the calculation below it will make the code simpler and easier to understand, and be less prone to errors.  ie:
XVectorF C;
XVectorF P(getos()->pos())
rad = (C-P).R();

Reply:
I think maybe the input variable names misled you on what the computation does. Your computation computes the distance between a combo hit and the circle center. I am computing the distance of closest approach to the circle. So I would need to pass two XYVectorF's (one for the point, and one for the circle), as well as the radius of the circle (which I was calling rC ... **xC, yC, rC stands for xCircle, yCircle, radiusCircle). I don't think this change you've suggested would make the code much simpler.